### PR TITLE
Add a url builder, resolve redirection TODOs, add breadcrumbs

### PR DESCRIPTION
**Pull request by yuxitang-amzn**
*Issue #, if available:*

*Description of changes:*
* Add a url builder
* Resolve redirection TODOs
* Add breadcrumbs

Screenshots for breadcrumbs:
Mockup:
image  

image

Implementation
image  

image

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: public/apps/configuration/utils/url-builder.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+ *   buildUrl(ResourceType.roles, Action.view, "readall") => "/roles/view/readall" (role detail page)
+ * edge case (wrong usage) fallbacks:
+ *   buildUrl(undefined, Action.new) => ""
+ *   buildUrl(ResourceType.roles, undefined, "readall") => "/roles"
+ */
+export function buildUrl(resouceType?: ResourceType, action?: Action, resourceId?: string) {
+  const rawContents = [resouceType, action, resourceId];
+  const contents = [];
+  for (const content of rawContents) {
+    if (content === undefined) {
```
Comment (tianleh) [inline public/apps/configuration/utils/url-builder.tsx:33]: Shall this be replaced by "!content" to handle more cases?

File: public/apps/configuration/utils/url-builder.tsx [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+import { ResourceType, Action } from '../types';
+
+/**
+ * Build hash based url.
+ * e.g.
+ *   buildUrl() => "/" (landing page)
+ *   buildUrl(ResourceType.roles) => "/roles" (role listing page)
+ *   buildUrl(ResourceType.roles, Action.new) => "/roles/create" (role creation page)
+ *   buildUrl(ResourceType.roles, Action.view, "readall") => "/roles/view/readall" (role detail page)
```
Comment (tianleh) [inline public/apps/configuration/utils/url-builder.tsx:24]: "readall" is a bit confusing as it sounds like read all the roles. Probably just call it aRole or sampleRole or logtash

File: public/apps/types.ts [hunk 3]
```diff
@@ -21,3 +21,7 @@ export interface AppDependencies {
   navigation: AppPluginStartDependencies;
   params: AppMountParameters;
 }
+
+export interface SingleResourcePageDependencies extends AppDependencies {
```
Comment (tianleh) [inline public/apps/types.ts:25]: Looks like BreadcrumbsPageDependencies is more close to what this interface performs.

File: public/apps/configuration/utils/url-builder.tsx (hunk 1 above)
Comment (yuxitang-amzn) [reply public/apps/configuration/utils/url-builder.tsx:33]: For null, I think typescript does not allow explicit null here. 
For empty string, I think we don't really need to block, there might be a case url with '//'.

File: public/apps/types.ts (hunk 3 above)
Comment (yuxitang-amzn) [reply public/apps/types.ts:25]: Good idea.

File: public/apps/configuration/utils/url-builder.tsx (hunk 2 above)
Comment (yuxitang-amzn) [reply public/apps/configuration/utils/url-builder.tsx:24]: Good idea.

**Pull request merged**